stl/inc/system_error

```diff
@@ -483,18 +485,14 @@ public:
     }
 
     _NODISCARD virtual string message(int _Errcode) const override {
-        const unsigned long _Size = 32767;
-        string _Narrow(_Size, '\0');
-
-        const unsigned long _Val = _Winerror_message(static_cast<unsigned long>(_Errcode), &_Narrow[0], _Size);
-        if (_Val == 0) {
-            _Narrow = "unknown error";
+        const char* _Msg     = nullptr;
+        const size_t _Length = _Winerror_message2(static_cast<unsigned long>(_Errcode), &_Msg);
+        if (_Length != 0) {

```

Comment:
I prefer the previous form that did not have a negation. (That is, if the if/else has an else, IMO one should avoid negations in the condition, because analyzing the else branch requires the reader to negate a negation)

Reply:
It's personal style to put the happy path first because compilers (until `[[likely]]` and `[[unlikely]]`) tend to assume the `if` case is more likely than the `else` case and generate code in a matter that is better for instruction fetching and decoding when hitting the `if` case, but I can adjust that.

Reply:
Resolved